src/query/models/types.go

```diff
@@ -150,3 +163,14 @@ type Metric struct {
 
 // Metrics is a list of individual metrics.
 type Metrics []Metric
+
+// Filters is a set of tag filters.
+type Filters []Filter
+
+// Filter is a regex tag filter.
+type Filter struct {
+	// Name is the name of the series.
+	Name []byte
+	// Filters are a set of regex expressions that filter matched tag values.
+	Filters []*regexp.Regexp
```

Comment:
Hm, perhaps we should also have some exact match ones here too (i.e. can specify either regexp or exact match)? There's cases where we just want exact match (in fact primarily now it's that) and ideally we wouldn't run regexp.

Reply:
Figured we'd handle this with anchored regex matchers?

Reply:
Yeah I'm just more concerned about if we only do exact matchers how much slower `bytes.Compare(.., ...) == 0` is vs `regexp.Match`?  Mainly for thousands of timeseries use case.